pcap-linux.c

```diff
@@ -3653,9 +3143,13 @@ init_tpacket(pcap_t *handle, int version, const char *version_str)
 			 * ENOPROTOOPT means the kernel is too old to
 			 * support PACKET_HDRLEN at all, which means
 			 * it either doesn't support TPACKET at all
-			 * or supports  only TPACKET_V1.
+			 * or supports only TPACKET_V1.
 			 */
-			return 1;	/* no */
+			pcap_fmt_errmsg_for_errno(handle->errbuf, PCAP_ERRBUF_SIZE, errno,
```

Comment:
I forgot that this is called with both TPACKET_V3 and TPACKET_V2, to try to get the best version of TPACKET, so it *does* need to have three different return values.

I think EINVAL means "we support TPACKET, we just don't support *that particular version of tpacket*", so that should cause it to return 1, so, if it was called with `TPACKET_V3`, `prepare_tpacket_socket()` can fall back on `TPACKET_V2`.